pkg/R/RDD.R

```diff
@@ -1374,6 +1374,48 @@ setMethod("top",
             takeOrderedElem(rdd, num, FALSE)
           })
 
+#' Pipes elements to a forked externel process.
+#'
+#' The same as `pipe()' in Spark.
+#'
+#' @param rdd The RDD whose elements are piped to the forked externel process.
+#' @param command the command to fork an externel process.
+#' @return a new RDD created by piping all elements to a forked externel process.
+#' @rdname pipeRDD
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, 1:10)
+#' collect(pipeRDD(rdd, "cat")
+#' Output: c("1", "2", ..., "10")
+#'}
+setGeneric("pipeRDD", function(rdd, command) { standardGeneric("pipeRDD") })
+
+#' @rdname pipeRDD
+#' @aliases pipeRDD,RDD,character-method
+setMethod("pipeRDD",
+          signature(rdd = "RDD", command = "character"),
+          function(rdd, command) {
+            func <- function(part) {
+              filename <- tempfile(pattern = "sparkr-pipe")
+              on.exit(unlink(filename))
+              pipeIn <- pipe(paste(command, ">", filename, sep=" "), "w")
```

Comment:
If we are going to go with output redirection, why not just use `system2` here ? We can set stdout to a variable in `system2` and stdin to the tempfile as you have now.